Fix: Redundant file fields for AnVIL are no longer needed (#7005)

dsotirho-ucsc opened the pull request:
<!--
This is the PR template for regular PRs against `develop`. Edit the URL in your
browser's location bar, appending either `&template=anvilprod-promotion.md`,
`&template=prod-promotion.md`, `&template=anvilprod-hotfix.md`, `&template=prod-
hotfix.md`, `&template=backport.md` or `&template=upgrade.md` to switch the
template.
-->

Linked issues: #7005

## Notes

- In order to minimize the time frame during which the service returns 500 responses, cancel the `deploy` stage and then be present when starting it manually, cancel the `integration_test` stage and immediately trigger the `early_reindex` when the `deploy_browser` stage finishes


## Checklist


### Author

- [x] PR is assigned to the author
- [x] Status of PR is *In progress*
- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] PR is linked to all issues it (partially) resolves
- [x] Status of linked issues is *In progress*
- [x] PR description links to linked issues
- [x] PR title matches<sup>1</sup> that of a linked issue <sub>or comment in PR explains why they're different</sub>
- [x] PR title references all linked issues
- [x] For each linked issue, there is at least one commit whose title references that issue

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title


### Author (partiality)

- [x] Added `p` tag to titles of partial commits
- [x] This PR is labeled `partial` <sub>or completely resolves all linked issues</sub>
- [x] This PR partially resolves each of the linked issues <sub>or does not have the `partial` label</sub>


### Author (reindex)

- [x] Added `r` tag to commit title <sub>or the changes introduced by this PR will not require reindexing of any deployment</sub>
- [x] This PR is labeled `reindex:dev` <sub>or the changes introduced by it will not require reindexing of `dev`</sub>
- [x] This PR is labeled `reindex:anvildev` <sub>or the changes introduced by it will not require reindexing of `anvildev`</sub>
- [x] This PR is labeled `reindex:anvilprod` <sub>or the changes introduced by it will not require reindexing of `anvilprod`</sub>
- [x] This PR is labeled `reindex:prod` <sub>or the changes introduced by it will not require reindexing of `prod`</sub>
- [x] This PR is labeled `reindex:partial` and its description documents the specific reindexing procedure for `dev`, `anvildev`, `anvilprod` and `prod` <sub>or requires a full reindex or carries none of the labels `reindex:dev`, `reindex:anvildev`, `reindex:anvilprod` and `reindex:prod`</sub>


### Author (API changes)

- [x] This PR and its linked issues are labeled `API` <sub>or this PR does not modify a REST API</sub>
- [x] Added `a` (`A`) tag to commit title for backwards (in)compatible changes <sub>or this PR does not modify a REST API</sub>
- [x] Updated REST API version number in `app.py` <sub>or this PR does not modify a REST API</sub>


### Author (upgrading deployments)

- [x] Ran `make docker_images.json` and committed the resulting changes <sub>or this PR does not modify `azul_docker_images`, or any other variables referenced in the definition of that variable</sub>
- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading deployments</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading deployments</sub>
- [x] This PR is labeled `upgrade` <sub>or does not require upgrading deployments</sub>
- [x] This PR is labeled `deploy:shared` <sub>or does not modify `docker_images.json`, and does not require deploying the `shared` component for any other reason</sub>
- [x] This PR is labeled `deploy:gitlab` <sub>or does not require deploying the `gitlab` component</sub>
- [x] This PR is labeled `deploy:runner` <sub>or does not require deploying the `runner` image</sub>


### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any linked issues <sub>or the none of the stable branches (`anvilprod` and `prod`) have temporary hotfixes for any of the issues linked to this PR</sub>


### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed fixups from prior reviews
- [x] Ran `make requirements_update` <sub>or this PR does not modify `Dockerfile`, `environment`, `requirements*.txt`, `common.mk`, `Makefile` or `environment.boot`</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not modify `requirements*.txt`</sub>
- [x] This PR is labeled `reqs` <sub>or does not modify `requirements*.txt`</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not modify functionality that could affect the IT outcome</sub>
- [x] PR is awaiting requested review from a peer
- [x] Status of PR is *Review requested*
- [x] PR is assigned to only the peer and the author


### Peer reviewer (after approval)

Note that after requesting changes, the PR must be assigned to only the author.

- [x] Actually approved the PR
- [x] PR is not a draft
- [x] PR is awaiting requested review from system administrator
- [x] Status of PR is *Review requested*
- [x] PR is assigned to only the system administrator and the author


### System administrator (after approval)

- [x] Actually approved the PR
- [x] Labeled linked issues as `demo` or `no demo`
- [x] Commented on linked issues about demo expectations <sub>or all linked issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] A comment to this PR details the completed security design review
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Status of PR is *Approved*
- [x] PR is assigned to only the operator and the author


### Operator

- [x] Checked `reindex:…` labels and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all linked issues are labeled `no demo`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub


### Operator (deploy `.shared` and `.gitlab` components)

- [x] Ran `_select dev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select dev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Ran `_select anvildev.shared && CI_COMMIT_REF_NAME=develop make -C terraform/shared apply_keep_unused` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.gitlab && CI_COMMIT_REF_NAME=develop make -C terraform/gitlab apply` <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Checked the items in the next section <sub>or this PR is labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the system administrator and the author <sub>or this PR is not labeled `deploy:gitlab`</sub>


### System administrator (post-deploy of `.gitlab` component)

- [x] Background migrations for [`dev.gitlab`](https://gitlab.dev.singlecell.gi.ucsc.edu/admin/background_migrations) are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] Background migrations for [`anvildev.gitlab`](https://gitlab.anvil.gi.ucsc.edu/admin/background_migrations) are complete <sub>or this PR is not labeled `deploy:gitlab`</sub>
- [x] PR is assigned to only the operator and the author


### Operator (deploy runner image)

- [x] Ran `_select dev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>
- [x] Ran `_select anvildev.gitlab && make -C terraform/gitlab/runner` <sub>or this PR is not labeled `deploy:runner`</sub>


### Operator (sandbox build)

- [x] Added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `dev`</sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices in `anvildev`</sub>
- [x] Started reindex in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR is not labeled `reindex:dev`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR is not labeled `reindex:anvildev`</sub>


### Operator (merge the branch)

- [x] All status checks passed and the PR is mergeable
- [x] The title of the merge commit starts with the title of this PR
- [x] Added PR # reference to merge commit title
- [x] Collected commit title tags in merge commit title <sub>but only included `p` if the PR is also labeled `partial`</sub>
- [x] Pushed merge commit to GitHub
- [x] Status of PR is *Merged lower*
- [x] Status of blocked issues is *Triage* <sub>or no issues are blocked on the linked issues</sub>


### Operator (main build)

- [x] Pushed merge commit to GitLab `dev`
- [x] Pushed merge commit to GitLab `anvildev`
- [x] Build passes on GitLab `dev`
- [x] Reviewed build logs for anomalies on GitLab `dev`
- [x] Build passes on GitLab `anvildev`
- [x] Reviewed build logs for anomalies on GitLab `anvildev`
- [x] Ran `_select dev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Ran `_select anvildev.shared && make -C terraform/shared apply` <sub>or this PR is not labeled `deploy:shared`</sub>
- [x] Deleted PR branch from GitHub
- [x] PR is assigned to only the operator
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`
- [x] Status of linked issues is *Lower*, or *Triage*, if PR is partial


### Operator (reindex)

- [x] Deindexed all unreferenced catalogs in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed all unreferenced catalogs in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Deindexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Deindexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Indexed specific sources in `dev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:dev`</sub>
- [x] Indexed specific sources in `anvildev` <sub>or this PR is neither labeled `reindex:partial` nor `reindex:anvildev`</sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Checked for, triaged and possibly requeued messages in both fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Emptied fail queues in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Emptied fail queues in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Restarted the Data Browser pipeline for the [ucsc/hca/dev branch](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/data-browser/-/pipelines/new?ref=ucsc%2Fhca%2Fdev) on GitLab in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Restarted the Data Browser pipeline for the [ucsc/lungmap/dev branch](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/data-browser/-/pipelines/new?ref=ucsc%2Flungmap%2Fdev) on GitLab in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Restarted `deploy_browser` job in the GitLab pipeline for this PR in `dev` <sub>or this PR does not require reindexing `dev`</sub>
- [x] Restarted the Data Browser pipeline for the [ucsc/anvil/anvildev branch](https://gitlab.anvil.gi.ucsc.edu/ucsc/data-browser/-/pipelines/new?ref=ucsc%2Fanvil%2Fanvildev) on GitLab in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>
- [x] Restarted `deploy_browser` job in the GitLab pipeline for this PR in `anvildev` <sub>or this PR does not require reindexing `anvildev`</sub>


### Operator (mirroring)

- [x] Started mirroring in `dev` <sub>or this PR does not require mirroring `dev`</sub>
- [x] Started mirroring in `anvildev` <sub>or this PR does not require mirroring `anvildev`</sub>
- [x] Checked for, triaged and possibly requeued messages in mirror fail queue in `dev` <sub>or this PR does not require mirroring `dev`</sub>
- [x] Checked for, triaged and possibly requeued messages in mirror fail queue in `anvildev` <sub>or this PR does not require mirroring `anvildev`</sub>
- [x] Emptied mirror fail queue in `dev` <sub>or this PR does not require mirroring `dev`</sub>
- [x] Emptied mirror fail queue in `anvildev` <sub>or this PR does not require mirroring `anvildev`</sub>


### Operator

- [x] Propagated the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `API`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels to the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] Propagated any specific instructions related to the `deploy:shared`, `deploy:gitlab`, `deploy:runner`, `API`, `reindex:partial`, `reindex:anvilprod` and `reindex:prod` labels, from the description of this PR to that of the next promotion PRs <sub>or this PR carries none of these labels</sub>
- [x] PR is assigned to no one


## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

dsotirho-ucsc committed:
[r] Remove files.uuid from AnVIL (#7005)

dsotirho-ucsc commented:
[7597_IT_2025-12-02.txt](https://github.com/user-attachments/files/23887245/7597_IT_2025-12-02.txt)

nadove-ucsc commented on src/azul/plugins/__init__.py:
```diff
@@ -159,6 +159,8 @@ class are more akin to a facet name, rather than a field name. However, not
     bundle_uuid: FieldName
     bundle_version: FieldName
+    file_uuid_facet: FieldName
+    file_uuid_field: FieldName
```
```suggestion
    file_id: FieldName
    file_id_facet: FieldName
```

nadove-ucsc replied:
None of the other entries here have a "_field" suffix, so I think this can be made more terse.

Given the disclaimers in this class's docstring, I suspect a more in-depth refactoring is needed here to resolve the tension between these two new fields. But I'm not going to speculate on what that refactoring should look like without @hannes-ucsc 's input.

nadove-ucsc reviewed (changes requested):
Looks good, just a naming recommendation 👍

dsotirho-ucsc commented:
[7597_IT_2025-12-03.txt](https://github.com/user-attachments/files/23914911/7597_IT_2025-12-03.txt)

hannes-ucsc commented on src/azul/plugins/metadata/anvil/__init__.py:
```diff
@@ -275,6 +274,8 @@ def special_fields(self) -> SpecialFields:
                              bundle_uuid='bundle_uuid',
                              bundle_version='bundle_version',
+                             file_id='files.document_id',
+                             file_id_facet='document_id',
```
One of these is misplaced. Instances of this class encapsulate the names of special *fields*, and fields are the names of properties in index documents. So based on that they can't both be fields. PL with Noa and you, please.

dsotirho-ucsc committed:
Remove unused special field root_entity_id

dsotirho-ucsc committed:
[r] Remove files.sha256 from AnVIL (#7005)

dsotirho-ucsc committed:
[r] Remove files.crc32c from AnVIL (#7005)

dsotirho-ucsc commented:
[7597_IT_2025-12-09.txt](https://github.com/user-attachments/files/24062382/7597_IT_2025-12-09.txt)

hannes-ucsc committed:
Promote type for inverse field mapping to top level

hannes-ucsc committed:
Fix typing of method, add docstring

hannes-ucsc committed:
Rename method

hannes-ucsc committed:
Add missing reference to special field

hannes-ucsc committed:
Implement abstract property with class attribute instead of property

hannes-ucsc committed:
Remove obvious comment

hannes-ucsc committed:
Remove redundant property override

hannes-ucsc committed:
Reorder methods

hannes-ucsc committed:
Add documentation to field-related types

hannes-ucsc committed:
Rename method

hannes-ucsc committed:
Extract properties for app and plugin used by tests

hannes-ucsc committed:
Protect test class attribute

hannes-ucsc committed:
Fix facet vs field misnomer

hannes-ucsc committed:
Remove redundant ABCMeta declaration

hannes-ucsc committed:
Push down DCP/1-specific method into DCP/1-specific subclass

hannes-ucsc reviewed (changes requested):
Please integration-test this with my changes in place.

dsotirho-ucsc commented:
The `test_indexing` test case failed during IT.

## Error from /repository/files

/aws/lambda/azul-service-daniel2

```
[INFO]	2025-12-11T21:59:01.274Z	3413813f-babc-441d-b600-a5c642ef9e86	azul.chalice	Received GET request for '/repository/files/7f91028d-5287-42ef-a8a6-68603c84387c', with {"query": {"catalog": "anvil-it", "version": "2022-06-01T00:00:00.000000Z"}, "headers": {"accept-encoding": "identity", "cloudfront-forwarded-proto": "https", "cloudfront-is-desktop-viewer": "true", "cloudfront-is-mobile-viewer": "false", "cloudfront-is-smarttv-viewer": "false", "cloudfront-is-tablet-viewer": "false", "cloudfront-viewer-asn": "7922", "cloudfront-viewer-country": "US", "host": "service.daniel2.anvil.gi.ucsc.edu", "user-agent": "python-urllib3/1.26.20", "via": "1.1 8cd8685b0adba2bf9ff8d080d6463d56.cloudfront.net (CloudFront)", "x-amz-cf-id": "bSX6wdmpGnzcsvqwJMmBpwGXzsFFOwvjazh9fZhAHC9CxQPP6zwWGg==", "x-amzn-trace-id": "Root=1-693b3ea5-1d2bfdf06b466cc039ba677a", "x-forwarded-for": "73.189.51.68, 18.68.29.19", "x-forwarded-port": "443", "x-forwarded-proto": "https"}}.
...
[ERROR]	2025-12-11T21:59:01.280Z	3413813f-babc-441d-b600-a5c642ef9e86	azul-service-daniel2	Caught exception for path /repository/files/{file_uuid}
Traceback (most recent call last):
  File "/var/task/chalice/app.py", line 1922, in _get_view_function_response
    response = view_function(**function_args)
  File "/var/task/app.py", line 1495, in repository_files
    result = _repository_files(file_uuid, fetch=False)
  File "/var/task/app.py", line 1548, in _repository_files
    return app.repository_controller.download_file(catalog=catalog,
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
                                                   fetch=fetch,
                                                   ^^^^^^^^^^^^
    ...<2 lines>...
                                                   headers=headers,
                                                   ^^^^^^^^^^^^^^^^
                                                   authentication=request.authentication)
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/service/repository_controller.py", line 234, in download_file
    file = self.service.get_data_file(catalog=catalog,
                                      file_uuid=file_uuid,
                                      file_version=file_version,
                                      filters=self.get_filters(catalog, authentication, None))
  File "/var/task/azul/service/repository_service.py", line 332, in get_data_file
    request = chain.prepare_request(request)
  File "/var/task/azul/service/elasticsearch_service.py", line 135, in prepare_request
    request = self.inner.prepare_request(request)
  File "/var/task/azul/service/elasticsearch_service.py", line 182, in prepare_request
    query = self.prepare_query()
  File "/var/task/azul/service/elasticsearch_service.py", line 231, in prepare_query
    for field_path, relation_and_values in self.prepared_filters.items():
                                           ^^^^^^^^^^^^^^^^^^^^^
  File "/var/task/azul/caching.py", line 203, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
                                               ~~~~~~~~~^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 196, in prepared_filters
    return self._translate_filters(filters_json)
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
  File "/var/task/azul/service/elasticsearch_service.py", line 219, in _translate_filters
    field = field_mapping[field]
            ~~~~~~~~~~~~~^^^^^^^
KeyError: 'file.uuid'
```

# Error from /manifests/files

/aws/lambda/azul-service-daniel2-manifest

```
[
    {
        "@timestamp": "2025-12-11 21:58:55.367",
        "@message": "START RequestId: a6926a90-1f35-40cc-9266-92e2eb4649ca Version: $LATEST\n"
    },
...
    {
        "@timestamp": "2025-12-11 21:58:55.410",
        "@message": "[INFO]\t2025-12-11T21:58:55.410Z\ta6926a90-1f35-40cc-9266-92e2eb4649ca\topensearch\tMaking POST request to https://vpc-azul-index-anvilbox-y7lvuxpx7jaryr6rpxubm6ey6u.us-east-1.es.amazonaws.com:443/azul_v2_daniel2_anvil-it_files_aggregate/_search"
    },
    {
        "@timestamp": "2025-12-11 21:58:55.410",
        "@message": "[INFO]\t2025-12-11T21:58:55.410Z\ta6926a90-1f35-40cc-9266-92e2eb4649ca\topensearch\t… with a request body of length 8781 starting in b'{\"query\":{\"bool\":{\"must\":[{\"constant_score\":{\"filter\":{\"range\":{\"contents.files.file_size\":{\"gte\":0,\"lte\":2001915174568,\"relation\":\"within\"}}}}},{\"constant_score\":{\"filter\":{\"terms\":{\"sources.id.keyword\":[\"001afc86-da4c-4739-85be-26ca98d2693f\",\"006f69b6-7f42-4eba-81df-86805c22dd3d\",\"01b1ac88-8144-428e-a84c-88e9ed8ec6a2\",\"02b9f4e0-3bee-4c23-87d4-485ddb414a87\",\"03b58060-2316-42fc-811f-0e5895399e96\",\"0522bcb4-e8f9-4c0b-936b-b2d729cbcac7\",\"0559efce-d188-4a59-9f6f-2a6258ca7e79\",\"057262af-09d8-4975-8647-2a68a5d33629\",\"05ed25dc-408d-4045-998d-6166fc2c3293\",\"06eca349-0e2c-4785-a93a-1eeb46a50220\",\"07bd9e9a-f3d5-4ce7-a7b5-a82cf5ede5b9\",\"0a4bd5cf-f007-49b2-98e7-f328490ede47\",\"0a711aa4-24d1-46e2-9d4b-5b02887311f9\",\"0ab78bd0-2e02-4536-b3c9-bdd7a57d7988\",\"0c86170e-312d-4b39-a0a4-2a2bfaa24c7a\",\"0e5501d1-39ca-44bd-b2a8-127841eb19ba\",\"104f2254-2af9-4699-9573-00d2da718430\",\"139ecf31-b6bb-419a-97c8-1245029631bb\",\"13c6bf33-41a3-4330-a3f7-1e69f675b01e\",\"15242c5c-0c11-441a-8b63-4cafcebe892f\",\"155a019c-ca47-432d-939b-6e51d5bec73...'"
    },
    {
        "@timestamp": "2025-12-11 21:58:55.442",
        "@message": "[INFO]\t2025-12-11T21:58:55.442Z\ta6926a90-1f35-40cc-9266-92e2eb4649ca\topensearch\tGot 200 response after 0.032s from POST to https://vpc-azul-index-anvilbox-y7lvuxpx7jaryr6rpxubm6ey6u.us-east-1.es.amazonaws.com:443/azul_v2_daniel2_anvil-it_files_aggregate/_search"
    },
    {
        "@timestamp": "2025-12-11 21:58:55.442",
        "@message": "[INFO]\t2025-12-11T21:58:55.442Z\ta6926a90-1f35-40cc-9266-92e2eb4649ca\topensearch\t… with a response body of length 309147 starting in '{\"took\":18,\"timed_out\":false,\"_shards\":{\"total\":1,\"successful\":1,\"skipped\":0,\"failed\":0},\"hits\":{\"total\":{\"value\":102,\"relation\":\"eq\"},\"max_score\":null,\"hits\":[{\"_index\":\"azul_v2_daniel2_anvil-it_files_aggregate\",\"_id\":\"0127949c-6bab-467e-b63a-2edeb0ceaf90\",\"_score\":null,\"_source\":{\"sources\":[{\"id\":\"790795c4-49b1-4ac8-a060-207b92ea08c5\",\"spec\":\"tdr:bigquery:gcp:datarepo-dev-e53e74aa:ANVIL_1000G_2019_Dev_20230609_ANV5_202306121732\"}],\"contents\":{\"activities\":[{\"assay_type\":[\"~null\"],\"activity_type\":[\"Checksum\",\"Unknown\"],\"activity_id\":[\"3d1e4ef4-9fb5-21a4-4061-22237cbffc02\",\"46c38de0-e993-40a4-39a1-9b9d82a294cf\",\"b8982983-66f0-c098-3774-4195c077912b\"],\"reference_assembly\":[\"~null\"],\"document_id\":[\"0fcec8ac-7e89-476a-9d7e-37373ee67162\",\"2318ac8b-9387-4649-bb01-d3450dc30409\",\"5208bdae-621a-4396-97ef-b621764ee811\"],\"data_modality\":[\"~null\"],\"source_datarepo_row_ids\":[\"file_inventory:3ccc05fe-b346-4286-bb96-02c51e0de295\",\"file_inventory:a09128fe-a603-4777-a502-11e1a734b835\",\"participant:9b20a5b4-ccca-47ff-a81a-b6…'"
    },
    {
        "@timestamp": "2025-12-11 21:58:55.448",
        "@message": "[ERROR] AssertionError: <class 'NoneType'>
Traceback (most recent call last):
  File "/var/task/azul/chalice.py", line 194, in patched_event_source_handler
    return old_handler(self_, event, context)
  File "/var/task/chalice/app.py", line 1761, in __call__
    return self.handler(event_obj)
  File "/var/task/chalice/app.py", line 1707, in __call__
    return self._original_func(event.to_dict(), event.context)
  File "/var/task/app.py", line 1355, in generate_manifest
    return app.manifest_controller.get_manifest(event)
  File "/var/task/azul/service/manifest_controller.py", line 96, in get_manifest
    result = self.service.get_manifest(format=manifest_key.format,
  File "/var/task/azul/service/manifest_service.py", line 642, in get_manifest
    return self._generate_manifest(generator, manifest_key, partition)
  File "/var/task/azul/service/manifest_service.py", line 651, in _generate_manifest
    partition = generator.write(manifest_key, partition)
  File "/var/task/azul/service/manifest_service.py", line 1329, in write
    partition = self.write_page_to(partition, output=text_buffer)
  File "/var/task/azul/service/manifest_service.py", line 1696, in write_page_to
    file['file_mirror_uri'] = self._azul_mirror_uri(file)
  File "/var/task/azul/service/manifest_service.py", line 1161, in _azul_mirror_uri
    file = self.metadata_plugin.file_class.from_index(file)
  File "/var/task/azul/plugins/metadata/anvil/__init__.py", line 549, in from_index
    md5=json_str(hit['file_md5sum']))
  File "/var/task/azul/types.py", line 165, in json_str
    return any_str(v)
  File "/var/task/azul/types.py", line 169, in any_str
    assert isinstance(v, str), type(v)"
    }
]
```

hannes-ucsc commented:
The manifest failures seem to be a combination of the most recent upgrade instructions not having been followed (`AZUL_ENABLE_MIRRORING` left at 0), and a bug in the code (manifest generation should work with AZUL_ENABLE_MIRRORING set to 0), a regression from the same PR that introduced the upgrade instructions. To confirm that hypothesis run IT on develop in your personal deployment with exactly the same configuration. If I am  right, the IT will fail with some of the error messages you post above. In that case, please open an issue with a minimal reproduction (just saying "run IT" is not minimal).

For the /repository/files failure, I will push a fix.

hannes-ucsc committed:
Allow for asymmetrically named special fields

…, i.e., special fields whose name in service response hits differs from the name used in facets, filters and the sort parameter of service requests

dsotirho-ucsc commented:
> To confirm that hypothesis run IT on develop in your personal deployment with exactly the same configuration. If I am right, the IT will fail with some of the error messages you post above. In that case, please open an issue with a minimal reproduction (just saying "run IT" is not minimal).

Rerunning IT from `develop` branch with an old `environment.py` (with mirroring disabled) did reproduce the manifest errors and did not produce the /repository/files errors. I created #7640 for the manifest errors.

hannes-ucsc commented:
I tested this in my personal AnVIL-based deployment. Please hold warm until  PR #7639 for #7638 lands.

dsotirho-ucsc commented:
[7597_IT_2025-12-12.txt](https://github.com/user-attachments/files/24138829/7597_IT_2025-12-12.txt)

hannes-ucsc commented:
### Security design review

- [x] Security design review completed; this PR does **not** …
  - [x] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with Terra 
  - [x] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [x] … affect the permissions of external users like access to
    - TDR snapshots
  - [x] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [x] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [x] … affect monitoring of the system
  - [x] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [x] … add an interface that exposes sensitive or confidential data at the security boundary
  - [x] … affect the encryption of data at rest
  - [x] … require persistence of sensitive or confidential data that might require encryption at rest
  - [x] … require unencrypted transmission of data within the security boundary
  - [x] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [x] Documentation on any unchecked boxes is provided in comments below
